src/logger.js

```diff
@@ -0,0 +1,60 @@
+
+export default class Logger {
+    constructor (options = {}) {
+        this._prefix = options.prefix;
+        this._transports = options.transports;
+        this._levels = new Map();
+        this._defaultFormat = options.defaultFormat;
+        this._createMethods(options.methods);
+    }
+
+    message (method, ...args) {
+        this._filterTransports(method)
+            .forEach(transport => {
+                var format = this._getFromatter(transport);
+                var formatted = format(method, this._prefix, args);
+                transport.write(method, formatted);
+            });
+    }
+
+    setLevels (transport, levels) {
```

Comment:
I wanted to make transports as simple as possible. Just objects with `format` and `write` methods. It's simpler to create new transports this way, without thinking about levels and stuff